Header header
int16[] map_array  
